src/foundation.cs

```diff
@@ -156,6 +156,16 @@ interface NSArray : NSSecureCoding, NSMutableCopying, INSFastEnumeration {
 
 		[Export ("removeObserver:fromObjectsAtIndexes:forKeyPath:")]
 		void RemoveObserver (NSObject observer, NSIndexSet indexes, string keyPath);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Export ("writeToURL:error:")]
+		bool WriteToUrl (NSUrl url, [NullAllowed] out NSError error);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Static]
+		[Export ("arrayWithContentsOfURL:error:")]
+		[return: NullAllowed]
+		NSObject[] FromUrl (NSUrl url, [NullAllowed] out NSError error);
```

Comment:
that's an odd case. In general returning an `[]` make sense, but this is a static method in `NSArray` so it should return an `NSArray` not another type.